File: commands/ticket_commands/openTickets.js
const { SlashCommandBuilder } = require('discord.js');
const { getAllTickets} = require('../../other_functions/ticketDatabaseFuncs.js');
const configJson = require('../../config.json');


module.exports = {
    data: new SlashCommandBuilder()
        .setName('open_tickets')
        .setDescription('Gets all the currently open tickets'),
    async execute(client, interaction) {
        const allowedRoleIds = configJson.adminRolesIDS; 
        if(!allowedRoleIds.some(roleId => interaction.member.roles.cache.has(roleId))){
            return interaction.reply({ content: 'No Permission', ephemeral: true });
        }

        const tickets = await getAllTickets();
        let message = [];
        for (let i = 0; i < tickets.length; i++) {
            const ticket = tickets[i];
            message.push(ticket.threadChannelID);
        }

        for (let i = 0; i < message.length; i++) {
            message[i] = `<#${message[i]}>`;
        }

        if (message.length === 0) {
            message.push('No tickets found.');
        }

        await interaction.reply({ content: message.join(' '), ephemeral: true });
    }
};

File: src/main/java/com/createciv/discord_bot/commands/ticket/OpenTickets.java
package com.createciv.discord_bot.commands.ticket;

import com.createciv.discord_bot.Bot;
import com.createciv.discord_bot.ConfigLoader;
import com.createciv.discord_bot.classes.SlashCommand;
import com.createciv.discord_bot.util.database.DatabaseRegistry;
import com.createciv.discord_bot.util.database.managers.TicketManager;
import com.createciv.discord_bot.util.database.types.TicketEntry;
import net.dv8tion.jda.api.JDA;
import net.dv8tion.jda.api.entities.Role;
import net.dv8tion.jda.api.entities.channel.Channel;
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;

import java.sql.SQLException;
import java.util.List;

public class OpenTickets extends SlashCommand {

    public OpenTickets() {
        super("open_tickets", "Gets all the currently open tickets");
    }

    @Override
    public void execute(SlashCommandInteractionEvent interactionEvent) {
        try {
            if(ConfigLoader.ADMIN_ROLE_IDS.isEmpty()){
                Bot.LOGGER.error("No admin roles selected. Please set this in the config to continue");
                interactionEvent.reply("Admin roles are not configured. Please configure them to use this command")
                        .setEphemeral(true).queue();
                return;
            }

            if(interactionEvent.getMember() == null) {
                interactionEvent.reply("Could not find member information").setEphemeral(true).queue();
                return;
            }

            boolean hasAdminRole = interactionEvent.getMember().getRoles().stream()
                    .map(Role::getId)
                    .anyMatch(ConfigLoader.ADMIN_ROLE_IDS::contains);

            if(!hasAdminRole){
                interactionEvent.reply("You do not have permission to run this command").setEphemeral(true).queue();
                return;
            }

            TicketManager manager = DatabaseRegistry.getTicketManager();
            List<TicketEntry> tickets;
            try {
                tickets = manager.getAllTickets();
            } catch (SQLException e) {
                throw new RuntimeException("Failed to fetch all tickets: " + e.getMessage());
            }

            if(tickets.isEmpty()){
                interactionEvent.reply("No tickets found").setEphemeral(true).queue();
                return;
            }

            StringBuilder message = new StringBuilder();
            JDA jda = interactionEvent.getJDA();
            for (TicketEntry ticket : tickets){
                Channel channel = jda.getThreadChannelById(ticket.threadChannelID);
                if(channel == null) throw new RuntimeException("Found missing thread in database. Please Fix");
                message.append(" <#").append(channel.getId()).append(">");
            }

            interactionEvent.reply(message.toString()).setEphemeral(true).queue();
        } catch (Exception e) {
            Bot.LOGGER.error("Error occurred in OpenTickets: {}", e.getMessage(), e);
            interactionEvent.reply("A fatal error occurred please try again later.");
            return;
        }
    }
}
